### PR TITLE
Feature/v1.0.0

### PR DESCRIPTION
**Pull request by sajadevo**
Templates are added

### CODE REVIEW

**Review by groovemen — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "react-dom": "^17.0.2",
+    "react-router-dom": "^5.2.0",
+    "react-scripts": "4.0.3",
+    "web-vitals": "^1.0.1"
+  },
+  "scripts": {
+    "start": "craco start",
+    "build": "craco build",
+    "test": "craco test",
+    "eject": "react-scripts eject"
```
Comment (einazare) [inline package.json]: Hey dude. I usually like to add one more script here like:
```
"install:clean": "rm -rf node_modules/ && rm -rf package-lock.json && npm install && npm start"
```
You could add two of them:
```
"npm:clean": "rm -rf node_modules/ && rm -rf package-lock.json && npm install && npm start",
"yarn:clean": "rm -rf node_modules/ && rm -rf package-lock.json && yarn install && yarn start"
```
For both, `yarn` and `npm`. Just a thought. Add them if you think they might be useful.

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="utf-8" />
+        <link rel="icon" href="./favicon.ico" />
+        <link
```
Comment (einazare) [inline public/index.html:6]: Hey dude, please move this from static stylesheets to node_modules:
- https://www.npmjs.com/package/@fortawesome/fontawesome-free

You can check out the usage of it in this project: 
- https://github.com/creativetimofficial/argon-dashboard-react
- https://github.com/creativetimofficial/argon-dashboard-react/blob/master/src/index.js

Comment (einazare) [reply public/index.html:6]: It is easier to maintain and update if they are in node_modules.

File: src/components/landing/Footer.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    return (
+        <>
+            <footer className="relative bg-gray-50 pt-8 pb-6">
+                <div className="container max-w-7xl mx-auto px-4">
+                    <div className="flex flex-wrap text-center lg:text-left pt-6">
+                        <div className="w-full lg:w-6/12 px-4">
+                            <H5 color="gray">Material Tailwind</H5>
+                            <LeadText
+                                color="blueGray"
+                                style={{ marginTop: '0' }}
```
Comment (einazare) [inline src/components/landing/Footer.js]: Hey dude. When I started to work with React, there were a lot of people that were not happy with using `style` (inline style). I think you could solve this by wrapping the `LeadText` component inside a `div` component with the class of `-mt-*`, the negative margin-top class of the normal one that `LeadText` uses.
Makes sense?

File: src/components/landing/Form.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                                <Input
+                                    type="email"
+                                    placeholder="Email Address"
+                                    color="lightBlue"
+                                />
+                            </div>
+
+                            <div
+                                className="relative w-full"
+                                style={{ height: '150px' }}
```
Comment (einazare) [inline src/components/landing/Form.js]: Once again, this `style` prop should be changed with one of the Tailwind CSS classes. You can choose between `h-36` and `h-40`, and if they are not ok, please create your own `h-*` class and use that one. For example `h-38` which would add a height of `9.5rem`, or `h-37` which would add a height of `9.25rem`.

File: src/components/landing/Form.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                            >
+                                <textarea
+                                    placeholder=" "
+                                    className="w-full h-full leading-normal shadow-none outline-none focus:outline-none focus:ring-0 px-0 md-input bg-transparent border-none md-input-light-blue-500"
+                                    style={{ resize: 'none' }}
```
Comment (einazare) [inline src/components/landing/Form.js]: You can use one of the classes from here: https://tailwindcss.com/docs/resize instead of the inline style

File: src/components/landing/Header.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import LeadText from '@material-tailwind/react/LeadText';
+
+export default function Header() {
+    const backgroundImage =
+        'https://images.unsplash.com/photo-1557804506-669a67965ba0?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1267&q=80';
+    return (
+        <div className="relative pt-16 pb-32 flex content-center items-center justify-center h-screen">
+            <div
+                className="absolute top-0 w-full h-full"
+                style={{
```
Comment (einazare) [inline src/components/landing/Header.js]: The background position and background sizes can be changed from inline style to tailwind css classes:
- https://tailwindcss.com/docs/background-position
- https://tailwindcss.com/docs/background-size
If they do not exist there, add them inside the tailwind config - i.e. extend the classes.

File: src/components/landing/Navbar.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+                                            <Icon
+                                                name="view_carousel"
+                                                size="2xl"
+                                                color="white"
+                                            />
+                                            &nbsp;Templates
+                                        </>
+                                    }
+                                    ripple="light"
+                                    style={{
```
Comment (einazare) [inline src/components/landing/Navbar.js]: Is there any way to change these inline styles to something else?

File: src/components/landing/StatusCard.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+import H6 from '@material-tailwind/react/Heading6';
+import Paragraph from '@material-tailwind/react/Paragraph';
+
+export default function StatusCard({ color, icon, title, children }) {
+    return (
+        <div className="w-full md:w-4/12 px-4 flex justify-center text-center">
+            <Card>
+                <CardBody>
+                    <div
+                        className={`text-white p-3 text-center inline-flex items-center justify-center w-12 h-12 mb-5 shadow-md rounded-full bg-${color}-500`}
```
Comment (einazare) [inline src/components/landing/StatusCard.js]: I don't really recommend using background classes like so, actually Tailwind CSS does not recommend using dynamic classes like so.
What we should do here, si we should have a constant with all colors that we want for this component:
```
const colors = {
    "red": "bg-red-500",
    "lightBlue": "bg-lightBlue-500",
    ... etc. you get the point
}
```
and after that, instead of `bg-${color}-500` we would say `${colors[color]}`

File: src/components/login/Navbar.js [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+                                        <Icon
+                                            name="view_carousel"
+                                            size="2xl"
+                                            color="white"
+                                        />
+                                        &nbsp;Templates
+                                    </>
+                                }
+                                ripple="light"
+                                style={{
```
Comment (einazare) [inline src/components/login/Navbar.js]: Same here as above, could we do something about this?

File: src/components/login/Page.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import BackgroundImage from '../../img/background-1920x1280.jpg';
+
+export default function Page({ children }) {
+    return (
+        <div
+            className="w-screen h-screen relative flex flex-col justify-between"
+            style={{
```
Comment (einazare) [inline src/components/login/Page.js]: Same as above with bgPosition and bgSize

File: src/components/profile/Footer.js [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+    return (
+        <>
+            <footer className="relative bg-gray-50 pt-8 pb-6">
+                <div className="container max-w-7xl mx-auto px-4">
+                    <div className="flex flex-wrap text-center lg:text-left pt-6">
+                        <div className="w-full lg:w-6/12 px-4">
+                            <H5 color="gray">Material Tailwind</H5>
+                            <LeadText
+                                color="blueGray"
+                                style={{ marginTop: '0' }}
```
Comment (einazare) [inline src/components/profile/Footer.js]: Same as above with negative margin top

File: src/components/profile/Header.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import BackgroundImage from '../../img/background-1920x1080.jpg';
+
+export default function Header() {
+    return (
+        <section className="relative block" style={{ height: '500px' }}>
+            <div
+                className="absolute top-0 w-full h-full"
+                style={{
```
Comment (einazare) [inline src/components/profile/Header.js]: Same with bgPosition and bgSize

File: src/components/profile/Navbar.js [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+                                            <Icon
+                                                name="view_carousel"
+                                                size="2xl"
+                                                color="white"
+                                            />
+                                            &nbsp;Templates
+                                        </>
+                                    }
+                                    ripple="light"
+                                    style={{
```
Comment (einazare) [inline src/components/profile/Navbar.js]: Same question as above with Dropdowns

**Review by einazare — CHANGES_REQUESTED** [review]
Also, beside all of the comments attached to the files, when doing a clean install (using npm), I have the following warnings. Is it something we could solve? I usually do:
```
npm i -E -O @babel/core typescript
```
To solve these warnings. If they cannot be solved, inside the CHANGELOG.md file, I add a warning about them, so that users may know these errors might appear on a clean install.
Screen Shot 2021-04-27 at 11 19 39 AM

Also, `npm start` does not work:

Screen Shot 2021-04-27 at 11 28 30 AM

I think you may have installed `craco` globally on your machine, and it is not added inside the package.json file, or something is missing either way from package.json file.

**Comment by sajadevo** [comment]
The main problem is with npm thats why npm start doesn't works but once we use the yarn install and run npm start its working fine I saw that lots of the modules are deprecated in npm and I will take a look to each of them to make it works.

As we as if we use the following command the npm start will work fine the same as yarn start
`rm -rf node_modules/ && rm -rf package-lock.json && npm install && npm start`

**Comment by einazare** [comment]
The idea is, that the majority of Creative Tim clients have been used to using `npm` instead of `yarn`, especially on the React products.
If you take a look at all of React products, you will see that, on the `Build Tools` pages, or `Getting started`, or `Installation`, we're saying that we're using `npm`. And I am afraid that, changing `npm` to `yarn` will cause issues.

So, if you want to keep `yarn`, that is fine, but we really need to make sure `npm` works as well.

**Comment by sajadevo** [comment]
No problem I want to have both there is no problem for me to use npm but having both option will be more better I will use both and make the npm to work as well.

**Comment by einazare** [comment]
I've edited the comments, it was a bit hard to follow the conversation 🤗

**Comment by sajadevo** [comment]
@EINazare the parts that you have mentioned are done and we also now have the latest version of tailwindcss running on the project you can check it and if you face with any issue lemme know.

File: src/components/landing/TeamSection.js [hunk 14]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import Title from 'components/landing/Title';
+import TeamCard from 'components/landing/TeamCard';
+import Image1 from '../../assets/img/team-1-800x800.jpg';
```
Comment (einazare) [inline src/components/landing/TeamSection.js]: Since you've used the jsconfig.json, you can use here absolute imports, i.e. 
```
import Image1 from 'assets/img/team-1-800x800.jpg';
```

File: src/components/landing/WorkingSection.js [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+import CardBody from '@material-tailwind/react/CardBody';
+import Icon from '@material-tailwind/react/Icon';
+import H4 from '@material-tailwind/react/Heading4';
+import H6 from '@material-tailwind/react/Heading6';
+import LeadText from '@material-tailwind/react/LeadText';
+import Paragraph from '@material-tailwind/react/Paragraph';
+import StatusCard from 'components/landing/StatusCard';
+
+export default function WorkingSection() {
+    const cardImageSrc =
```
Comment (einazare) [inline src/components/landing/WorkingSection.js]: Should we download these images and have them inside the product? This is useful if the owner of the image deletes it, right?

File: src/components/profile/Content.js [hunk 16]
```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import Button from '@material-tailwind/react/Button';
+import Image from '@material-tailwind/react/Image';
+import H3 from '@material-tailwind/react/Heading3';
+import Icon from '@material-tailwind/react/Icon';
+import LeadText from '@material-tailwind/react/LeadText';
+import ProfilePicture from '../../assets/img/team-2-800x800.jpg';
```
Comment (einazare) [inline src/components/profile/Content.js]: Once again, use absolute imports?

File: src/components/profile/Header.js [hunk 17]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+export default function Header() {
+    return (
+        <section className="relative block" style={{ height: '500px' }}>
```
Comment (einazare) [inline src/components/profile/Header.js]: Could we use something like 
```
className="relative block h-96"
```
instead of the inline style? Or, we could add a new class named `h-125` which would add `height: 31.25rem`:
```
```
className="relative block h-125"
```
And somewhere inside the tailwind config we would have:
```
"125": "31.25rem"
```
31.25rem = 500px (if base is 16).

File: src/index.js [hunk 18]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { BrowserRouter } from 'react-router-dom';
+import 'assets/styles/tailwind.css';
```
Comment (einazare) [inline src/index.js]: Hmmm, we have this style import here, and we have another on inside `src/App.js`. Could we add them all here?

File: tailwind.config.js [hunk 19]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
+    mode: 'jit',
+    purge: ['./src/**/*.js', './src/**/**/*,js', './public/index.html'],
+    darkMode: false, // or 'media' or 'class'
+    theme: {
+        extend: {
+            backgroundImage: () => ({
```
Comment (einazare) [inline tailwind.config.js:12]: Niiiiiiiiiiiice <3, I realy like this.

File: src/components/landing/Navbar.js [hunk 20]
```diff
@@ -0,0 +1,128 @@
...
+                                                    color="white"
+                                                />
+                                                <span className="ml-2">
+                                                    Templates
+                                                </span>
+                                            </div>
+                                        }
+                                        ripple="light"
+                                    >
+                                        <Link to="/landing">
```
Comment (einazare) [inline src/components/landing/Navbar.js]: Since this page is the same with the one from `/`, lets keep just one of them, either `/`, either `/landing`

File: src/App.js [hunk 21]
```diff
@@ -0,0 +1,27 @@
...
+// Font Awesome Style Sheet
+import '@fortawesome/fontawesome-free/css/all.min.css';
+
+// Material Tailwind Style Sheet
+import '@material-tailwind/react/tailwind.css';
+
+function App() {
+    return (
+        <Switch>
+            <Route exact path="/" component={Landing} />
```
Comment (einazare) [inline src/App.js:19]: Lets keep just one of the 
```
<Route exact path="/" component={Landing} />
<Route exact path="/landing" component={Landing} />
```

**Review by einazare — CHANGES_REQUESTED** [review]
`src/components/login/Footer.js` and `src/components/profile/Footer.js` should be merged into just one - this is beacuse the 2 components are exactly the same
`src/components/profile/Navbar.js` and `src/components/login/Navbar.js` and `src/components/landing/Navbar.js` should be merged into just one - this is beacuse the 3 components are exactly the same
We have some errors inside the console:
Screen Shot 2021-05-04 at 2 27 03 PM
Screen Shot 2021-05-04 at 2 26 46 PM
Screen Shot 2021-05-04 at 2 14 52 PM

**Comment by sajadevo** [comment]
The console errors are with the material tailwind it self and will be fixed on the new update, I'm currently working on it.

**Comment by einazare** [comment]
One of the errors comes from here:
```
<NavbarBrand>
                            <a
                                href="https://material-tailwind.com?ref=mtk"
                                target="_blank"
                                rel="noreferrer"
                            >
                                Material Tailwind
                            </a>
                        </NavbarBrand>
```
The ideea is that `NavbarBrand` is an anchor tag, while inside it, you add another `<a>` tag. Makes sense?

**Comment by sajadevo** [comment]
Thats what I said in the new update of material tailwind I will add a rest prop for the navbar brand to make it possible for the user to add a href to the navbar brand.

**Review by groovemen — CHANGES_REQUESTED** [review]
There is another font (or the Roboto Slab is not applied). You can compare with our Material Tailwind Landing template [here](https://material-tailwind.com/templates/landing). 

Screenshot 2021-05-05 at 12 08 20Screenshot 2021-05-05 at 12 08 33

Also, there is more space in this section (beside different fonts):

Screenshot 2021-05-05 at 12 13 33Screenshot 2021-05-05 at 12 13 43

Otherwise everything seems fine!

**Comment by sajadevo** [comment]
> There is another font (or the Roboto Slab is not applied). You can compare with our Material Tailwind Landing template [here](https://material-tailwind.com/templates/landing).
> 
> Screenshot 2021-05-05 at 12 08 20Screenshot 2021-05-05 at 12 08 33
> 
> Also, there is more space in this section (beside different fonts):
> 
> Screenshot 2021-05-05 at 12 13 33Screenshot 2021-05-05 at 12 13 43
> 
> Otherwise everything seems fine!

Fixed the fonts are added and the spacing is fixed as well.

**Review by groovemen — APPROVED** [review]
